statsmodels/sandbox/stats/diagnostic.py

```diff
@@ -1001,6 +1001,85 @@ def linear_lm(resid, exog, func=None):
     lm_pval = stats.chi2.sf(lm, k_vars - 1)
     return lm, lm_pval, ftest
 
+
+def spec_white(resid, exog):
+    '''
+    White's Two-Moment Specification Test
+
+    Parameters
+    ----------
+    resid : array_like
+        OLS residuals
+    exog : array_like
+        OLS design matrix
+
+    Returns
+    -------
+    stat : float
+        test statistic
+    pval : float
+        chi-square p-value for test statistic
+    dof : int
+        degrees of freedom
+
+    Notes
+    -----
+    Implements the two-moment specification test described by White's
+    Theorem 2 (1980, p. 823) which compares the standard OLS covariance
+    estimator with White's heteroscedasticity-consistent estimator. The
+    test statistic is shown to be chi-square distributed.
+
+    Null hypothesis is homoscedastic and correctly specified.
+
+    Assumes the OLS design matrix contains an intercept term and at least
+    one variable. The intercept is removed to calculate the test statistic.
+
+    Interaction terms (squares and crosses of OLS regressors) are added to
+    the design matrix to calculate the test statistic.
+
+    Degrees-of-freedom (full rank) = nvar + nvar * (nvar + 1) / 2
+
+    Linearly dependent columns are removed to avoid singular matrix error.
+
+    Reference
+    ---------
+    White, H. (1980). A heteroskedasticity-consistent covariance matrix
+    estimator and a direct test for heteroscedasticity. Econometrica,
+    48: 817-838.
```

Comment:
See also for `het_white`

Reply:
@bashtage @josef-pkt - Sorry for the delay. I made the changes above except for relocating out of sandbox. I believe the reason it's in sandbox is due to the still experimental nature of the collinearity check. Is this correct?

Also, there's been no further comments regarding the Zivot-Andrews test. Is there still interest in my generating a PR for that? See https://github.com/statsmodels/statsmodels/issues/4746